File: euler/0001/go/solution.go
package main

import "fmt"

func calculate_sum(n int) uint64 {
	var sum uint64 = 0

	for ; n > 0; n-- {
		if n % 3 == 0 || n % 5 == 0 {
			sum = sum + uint64(n)
		}
	}
	return sum;
}

func main() {
	fmt.Println(calculate_sum(1000))
}

File: euler/0001/rust/solution.rs

fn calculate_sum(mut limit: u32) -> u64
{
    let mut sum: u64 = 0;
    
    while limit > 0 {
        if limit % 3 == 0 || limit % 5 == 0 {
            sum = sum + limit as u64;
        }
        limit -= 1;
    }
    return sum;
}

fn main()
{
    let limit: u32 = 1000;
    println!("{}", calculate_sum(limit));
}
